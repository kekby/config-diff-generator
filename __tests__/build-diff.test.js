import path from 'path';
import { describe, expect } from '@jest/globals';

import buildDiff from '../src';

const getFixturePath = (filename) => path.join(__dirname, '..', '__tests__/fixtures', filename);

describe('generates diff', () => {
  test('works with json', () => {
    const json1 = getFixturePath('file1.json');
    const json2 = getFixturePath('file2.json');
    expect(buildDiff(json1, json2)).toMatchSnapshot();
  });

  test('json plain format', () => {
    const json1 = getFixturePath('file1.json');
    const json2 = getFixturePath('file2.json');
    expect(buildDiff(json1, json2, 'plain')).toMatchSnapshot();
  });

  test('format to json', () => {
    const json1 = getFixturePath('file1.json');
    const json2 = getFixturePath('file2.json');
    expect(buildDiff(json1, json2, 'json')).toMatchSnapshot();
  });

  test('works with yml', () => {
    const yml1 = getFixturePath('file1.yml');
    const yml2 = getFixturePath('file2.yml');
    expect(buildDiff(yml1, yml2)).toMatchSnapshot();
  });

  test('open existing file should log error message', () => {
    expect(() => buildDiff('doesnotexist.json', 'doesnt matter')).toThrow();
  });
});
