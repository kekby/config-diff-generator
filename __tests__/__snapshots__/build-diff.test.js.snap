// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates diff format to json 1`] = `"[{\\"name\\":\\"common\\",\\"state\\":\\"nested\\",\\"children\\":[{\\"name\\":\\"follow\\",\\"state\\":\\"added\\",\\"value\\":false},{\\"name\\":\\"setting1\\",\\"state\\":\\"unchanged\\",\\"value\\":\\"Value 1\\"},{\\"name\\":\\"setting2\\",\\"state\\":\\"removed\\",\\"value\\":200},{\\"name\\":\\"setting3\\",\\"state\\":\\"updated\\",\\"oldValue\\":true,\\"newValue\\":null},{\\"name\\":\\"setting4\\",\\"state\\":\\"added\\",\\"value\\":\\"blah blah\\"},{\\"name\\":\\"setting5\\",\\"state\\":\\"added\\",\\"value\\":{\\"key5\\":\\"value5\\"}},{\\"name\\":\\"setting6\\",\\"state\\":\\"nested\\",\\"children\\":[{\\"name\\":\\"doge\\",\\"state\\":\\"nested\\",\\"children\\":[{\\"name\\":\\"wow\\",\\"state\\":\\"updated\\",\\"oldValue\\":\\"\\",\\"newValue\\":\\"so much\\"}]},{\\"name\\":\\"key\\",\\"state\\":\\"unchanged\\",\\"value\\":\\"value\\"},{\\"name\\":\\"ops\\",\\"state\\":\\"added\\",\\"value\\":\\"vops\\"}]}]},{\\"name\\":\\"group1\\",\\"state\\":\\"nested\\",\\"children\\":[{\\"name\\":\\"baz\\",\\"state\\":\\"updated\\",\\"oldValue\\":\\"bas\\",\\"newValue\\":\\"bars\\"},{\\"name\\":\\"foo\\",\\"state\\":\\"unchanged\\",\\"value\\":\\"bar\\"},{\\"name\\":\\"nest\\",\\"state\\":\\"updated\\",\\"oldValue\\":{\\"key\\":\\"value\\"},\\"newValue\\":\\"str\\"}]},{\\"name\\":\\"group2\\",\\"state\\":\\"removed\\",\\"value\\":{\\"abc\\":12345,\\"deep\\":{\\"id\\":45}}},{\\"name\\":\\"group3\\",\\"state\\":\\"added\\",\\"value\\":{\\"deep\\":{\\"id\\":{\\"number\\":45}},\\"fee\\":100500}}]"`;

exports[`generates diff json plain format 1`] = `
"Property 'common.follow' was added with value: false
Property 'common.setting2' was removed
Property 'common.setting3' was updated. From true to null
Property 'common.setting4' was added with value: 'blah blah'
Property 'common.setting5' was added with value: [complex value]
Property 'common.setting6.doge.wow' was updated. From '' to 'so much'
Property 'common.setting6.ops' was added with value: 'vops'
Property 'group1.baz' was updated. From 'bas' to 'bars'
Property 'group1.nest' was updated. From [complex value] to 'str'
Property 'group2' was removed
Property 'group3' was added with value: [complex value]"
`;

exports[`generates diff works with json 1`] = `
"{
    common: {
      + follow: false
        setting1: Value 1
      - setting2: 200
      - setting3: true
      + setting3: null
      + setting4: blah blah
      + setting5: {
            key5: value5
        }
        setting6: {
            doge: {
              - wow: 
              + wow: so much
            }
            key: value
          + ops: vops
        }
    }
    group1: {
      - baz: bas
      + baz: bars
        foo: bar
      - nest: {
            key: value
        }
      + nest: str
    }
  - group2: {
        abc: 12345
        deep: {
            id: 45
        }
    }
  + group3: {
        deep: {
            id: {
                number: 45
            }
        }
        fee: 100500
    }
}"
`;

exports[`generates diff works with yml 1`] = `
"{
    common: {
      + follow: false
        setting1: Value 1
      - setting2: 200
      - setting3: true
      + setting3: null
      + setting4: blah blah
      + setting5: {
            key5: value5
        }
        setting6: {
            doge: {
              - wow: 
              + wow: so much
            }
            key: value
          + ops: vops
        }
    }
    group1: {
      - baz: bas
      + baz: bars
        foo: bar
      - nest: {
            key: value
        }
      + nest: str
    }
  - group2: {
        abc: 12345
        deep: {
            id: 45
        }
    }
  + group3: {
        deep: {
            id: {
                number: 45
            }
        }
        fee: 100500
    }
}"
`;
